File: curriculum/code/ruby/beginner/dog_years/dog_years_spec.rb
require './dog_years'

describe Dog, "#age_in_years" do

  it "should return an Integer" do
    Dog.new.age_in_years.should be_a_kind_of(Integer)
  end
end

describe Dog, "#age_in_days" do
  let(:dog) { Dog.new }

  it "should return the dog's age in years time 365" do
    dog.age_in_days.should eq(dog.age_in_years * 365)
  end

end

describe Dog, "#age_in_dog_years" do
  let(:dog) { Dog.new }

  it "should return the dog's age in years times 7" do
    dog.age_in_dog_years.should eq(dog.age_in_years * 7)
  end
  
end

File: curriculum/code/javascript/quizzes/beginner/dog_years/spec/DogYearsSpec.js
describe("Dog", function () {
  dog = new Dog();

  describe("#age", function() {
    it("should return an Integer", function(){
      expect(typeof dog.age).toEqual('number');
    });
  });

  describe("#age_in_years", function() {
    it("should return an Integer", function(){
      expect(typeof dog.age_in_years()).toEqual('number');
    });
  });

  describe("#age_in_days", function() {
    it("should return the dog's age in years times 365", function(){
      expect(dog.age_in_days()).toEqual(dog.age_in_years() * 365);
    });
  });

  describe("#age_in_dog_years", function() {
    it("should return the dog's age in years times 7", function(){
      expect(dog.age_in_dog_years()).toEqual(dog.age_in_years() * 7);
    });
  });

});